openedx/core/djangoapps/waffle_utils/views.py

```diff
@@ -214,20 +217,53 @@ def _get_or_create_toggle_response(self, toggles_dict, toggle_name):
 
     def _get_settings_state(self):
         """
-        Returns a dictionary of settings values. Will only return values that are set to true or false.
+        Returns a dictionary of settings values. Will only return values that are set to true or false and for which a
+        corresponding SettingToggle or SettingDictToggle does not exist.
         """
+        setting_toggles = set(toggle.name for toggle in SettingToggle.get_instances())
+        setting_dict_toggles = set((toggle.name, toggle.key) for toggle in SettingDictToggle.get_instances())
```

Comment:
**Note:** This comment relates to my earlier comment.  
1. For consistency, this function could instead be `def _add_setting_state(self, settings_dict):`.
2. This function could use `get_or_create_toggle_response`.  I'll leave this up to you.
3. If you call this function before the function for `SettingToggle` or `SettingDictToggle`, you don't need to check `settings_dict` (the new argument to this function), because it couldn't be in there yet.

Reply:
I'll take your comments into account in the later PR that adds the setting toggles to the endpoint. Just to clarify: we agree that SettingToggle or SettingDictToggle should override settings in the `response['django_settings']`, right? It was my understanding that we should not keep entries with duplicate names, even if they have different implementations (as indicated in the new unit tests).

Reply:
We can discuss more in the follow-up PR.  Can you point back to this review's comments from that PR description?

> Just to clarify: we agree that SettingToggle or SettingDictToggle should override settings in the response['django_settings'], right? 

I'm not sure what you mean by override.  It should enhance any existing data.  If it is overriding AND disagreeing with any earlier data, that shouldn't happen (I think), and we'd want to add some sort of warning field to let us know what is going on.

> It was my understanding that we should not keep entries with duplicate names, even if they have different implementations (as indicated in the new unit tests).

Agreed.  Again, we could always append to some warning field if we find anything odd going on.  However, I'm not worries about Setting and SettingDicts having the same name.  I'd be more worried about something like two SettingToggles defined with different default or a different module, etc.